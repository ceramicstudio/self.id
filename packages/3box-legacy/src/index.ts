/**
 * Load legacy 3Box profiles.
 *
 * ## Purpose
 *
 * The`3box-legacy` module provides utility functions to load legacy 3Box profiles.
 *
 * ## Installation
 *
 * ```sh
 * npm install @self.id/3box-legacy
 * ```
 *
 * ## Common use-cases
 *
 * ### Load a legacy 3Box profile as a Self.ID basic profile
 *
 * ```ts
 * import { getLegacy3BoxProfileAsBasicProfile } from '@self.id/3box-legacy'
 *
 * async function getProfile() {
 *   // Ethereum address to load the profile of
 *   const address = '0x123efad...'
 *   return await getLegacy3BoxProfileAsBasicProfile(address)
 * }
 * ```
 *
 * @module 3box-legacy
 */

import type { BasicProfile } from '@datamodels/identity-profile-basic'
import crossFetch from 'cross-fetch'

export async function loadLegacy3BoxProfile<T = Record<string, any>>(
  address: string,
  fetchFunc: typeof crossFetch = crossFetch
): Promise<T | null> {
  try {
    const res = await fetchFunc(`https://ipfs.3box.io/profile?address=${address}`)
    return res.ok ? ((await res.json()) as T) : null
  } catch (err) {
    return null
  }
}

// Validation for BasicProfile
const lengthIndex: Record<string, number> = {
  name: 150,
  description: 420,
  location: 140, //homeLocation
  website: 240, // url
  emoji: 2,
  employer: 140, //affiliations
  school: 140, //affiliations
}

const isStrAndLen = (obj: Record<string, any>, key: string): boolean => {
  if (!lengthIndex[key]) return false
  return typeof obj[key] === 'string' && (obj[key] as string).length <= lengthIndex[key]
}

export const transformProfile = (profile: Record<string, any>): BasicProfile => {
  const transform = {} as BasicProfile
  let image, background
  if (isStrAndLen(profile, 'name')) transform.name = profile.name as string
  if (isStrAndLen(profile, 'description')) transform.description = profile.description as string
  if (isStrAndLen(profile, 'location')) transform.homeLocation = profile.location as string
  if (isStrAndLen(profile, 'website')) transform.url = profile.website as string
  if (isStrAndLen(profile, 'emoji')) transform.emoji = profile.emoji as string
  if (isStrAndLen(profile, 'employer')) transform.affiliations = [profile.employer]
  if (isStrAndLen(profile, 'school')) {
    transform.affiliations = (transform.affiliations || []).concat([profile.school])
  }
  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
  if (Array.isArray(profile.image)) image = profile.image[0]?.contentUrl['/'] as string
  if (image != null && typeof image === 'string') {
    transform.image = {
      original: {
        src: `ipfs://${image}`,
        mimeType: 'application/octet-stream',
        width: 170,
        height: 170,
      },
    }
  }
  if (Array.isArray(profile.coverPhoto)) {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    background = profile.coverPhoto[0]?.contentUrl['/'] as string
  }
  if (background != null && typeof background === 'string') {
    transform.background = {
      original: {
        src: `ipfs://${background}`,
        mimeType: 'application/octet-stream',
        width: 1000,
        height: 175,
      },
    }
  }
  return transform
}

export async function getLegacy3BoxProfileAsBasicProfile(
  address: string,
  fetchFunc?: typeof crossFetch
): Promise<BasicProfile | null> {
  const profile = await loadLegacy3BoxProfile(address, fetchFunc)
  return profile ? transformProfile(profile) : null
}
