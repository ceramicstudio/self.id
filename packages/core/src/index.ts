/**
 * Read public records in Node and browsers environments.
 *
 * ## Purpose
 *
 * The `core` module of the Self.ID SDK exports the `Core` and `PublicID` classes to provide APIs
 * for reading public records in Node and browsers environments.
 *
 * It can therefore be used both client-side and server-side by applications wanting to access
 * records associated to a DID on a Ceramic network.
 *
 * ## Installation
 *
 * ```sh
 * npm install @self.id/core
 * ```
 *
 * ## Common use-cases
 *
 * ### Read the basic profile of a known DID
 *
 * ```ts
 * import { Core } from '@self.id/core'
 *
 * const core = new Core({ ceramic: 'testnet-clay' })
 *
 * async function getProfile() {
 *   // Replace by the wanted DID string
 *   const did = 'did:3:...'
 *   return await core.get('basicProfile', did)
 * }
 * ```
 *
 * ### Connect to a custom Ceramic node
 *
 * ```ts
 * import { Core } from '@self.id/core'
 *
 * // You can provide the URL of a custom node as the `ceramic` parameter
 * const core = new Core({ ceramic: 'https://ceramic-clay.3boxlabs.com' })
 * ```
 *
 * ### Use a custom data model
 *
 * ```ts
 * import { Core } from '@self.id/core'
 *
 * const model = {
 *   definitions: {
 *     basicProfile: 'kjzl6cwe1jw145cjbeko9kil8g9bxszjhyde21ob8epxuxkaon1izyqsu8wgcic',
 *     notes: 'kjzl6cwe1jw14688v7zhf5eo1uiblt0thkgdedb5gaiq99isj224g6v8uza2r4m',
 *   },
 *   schemas: {
 *     BasicProfile: 'ceramic://k3y52l7qbv1frxt706gqfzmq6cbqdkptzk8uudaryhlkf6ly9vx21hqu4r6k1jqio',
 *     Note: 'ceramic://k3y52l7qbv1fryg3uy4gz50pj3ivni18h6younsgd2kb5u7mivx3h258m01tkwk5c',
 *     Notes: 'ceramic://k3y52l7qbv1frxk024s7ds2456qo3rwa02k31uf98loz8lm1lodlgv3eeqh4cxgjk',
 *   },
 *   tiles: {
 *     placeholderNote: 'kjzl6cwe1jw148snra9bmyf32y5pj2ysnvenqx59k7n2lz3gnri72axvp25fo1v',
 *   },
 * }
 * const core = new Core({ ceramic: 'testnet-clay', model })
 *
 * async function getNotes(did) {
 *   return await core.get('notes', did)
 * }
 * ```
 *
 * ### Use a PublicID instance
 *
 * ```ts
 * import { Core, PublicID } from '@self.id/core'
 *
 * const core = new Core({ ceramic: 'testnet-clay' })
 * // Replace the `id` parameter by the wanted DID string
 * const currentUser = new PublicID({ core, id: 'did:3:...' })
 *
 * async function getCurrentUserProfile() {
 *   return await currentUser.get('basicProfile')
 * }
 * ```
 *
 * @module core
 */

export * from './core'
export * from './public'
export * from './types'
export * from './utils'
