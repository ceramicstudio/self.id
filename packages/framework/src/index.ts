/**
 * React framework for decentralized apps.
 *
 * ## Purpose
 *
 * The `framework` module is the highest-level abstraction provided by the Self.ID SDK, aimed at
 * helping developers to quickly get started with building decentralized apps using Ceramic with
 * React.
 *
 * The framework is built on top of the {@linkcode core}, {@linkcode web} and {@linkcode react}
 * modules to provide APIs to easily authenticate users, keep track of the current user and
 * interact with both public (read-only) and user-owned (mutable) records.
 *
 * ## Installation
 *
 * ```sh
 * npm install @self.id/framework
 * ```
 *
 * ## Common use-cases
 *
 * ### Configure the Provider
 *
 * The {@linkcode react.Provider Provider} component must be added at the root of the
 * application tree in order to use the hooks described below. It can be used to provide a custom
 * configuration for the Self.ID clients, authentication, state and UI options.
 *
 * ```ts
 * import { Provider } from '@self.id/framework'
 *
 * function App({ children }) {
 *   return <Provider client={{ ceramic: 'testnet-clay' }}>{children}</Provider>
 * }
 * ```
 *
 * ### Authenticate the user
 *
 * The framework provides a React hook to easily initiate an authentication flow for the Viewer
 * (the "current user" of the app). This flow is made of the following steps:
 *
 * The user authentication flow consists of the following steps:
 *
 * 1. An [Ethereum authentication provider](https://developers.ceramic.network/reference/typescript/classes/_ceramicnetwork_blockchain_utils_linking.ethereumauthprovider-1.html) is created using the Ethereum provider.
 * 1. The auth flow with 3ID Connect starts, using the [Ethereum authentication provider](https://developers.ceramic.network/reference/typescript/classes/_ceramicnetwork_blockchain_utils_linking.ethereumauthprovider-1.html).
 * 1. A {@linkcode web.SelfID SelfID} instance is created and stored in application state.
 *
 * Once this flow is successfully applied, the Viewer's cookie is set to the authenticated DID and
 * writing records associated to the Viewer becomes possible.
 *
 * ```ts
 * import { useViewerConnection } from '@self.id/framework'
 *
 * // A simple button to initiate the connection flow. A Provider must be present at a higher level
 * // in the component tree for the `useViewerConnection()` hook to work.
 * function ConnectButton() {
 *   const [connection, connect, disconnect] = useViewerConnection()
 *
 *   return connection.status === 'connected' ? (
 *     <button
 *       onClick={() => {
 *         disconnect()
 *       }}>
 *       Disconnect ({connection.selfID.id})
 *     </button>
 *   ) : 'ethereum' in window ? (
 *     <button
 *       disabled={connection.status === 'connecting'}
 *       onClick={async () => {
 *         const accounts = await window.ethereum.request({
 *           method: 'eth_requestAccounts',
 *         })
 *         await connect(new EthereumAuthProvider(window.ethereum, accounts[0]))
 *       }}>
 *       Connect
 *     </button>
 *   ) : (
 *     <p>
 *       An injected Ethereum provider such as{' '}
 *       <a href="https://metamask.io/">MetaMask</a> is needed to authenticate.
 *     </p>
 *   )
 * }
 * ```
 *
 * ### Interact with a viewer record
 *
 * The {@linkcode react.useViewerRecord useViewerRecord} hook loads the record for a given
 * definition in the index of the current viewer, with the following variants:
 *
 * - If no viewer is set, no record can be loaded
 * - If the viewer is not authenticated, the record gets loaded but cannot be mutated
 * - If the viewer is authenticated, the record gets loaded and be mutated
 *
 * ```ts
 * import { useViewerRecord } from '@self.id/framework'
 *
 * // Load and display the record contents
 * function ShowViewerName() {
 *   const record = useViewerRecord('basicProfile')
 *
 *   const text = record.isLoading
 *     ? 'Loading...'
 *     : record.content
 *     ? `Hello ${record.content.name || 'stranger'}`
 *     : 'No profile to load'
 *   return <p>{text}</p>
 * }
 *
 * // Mutate the record
 * function SetViewerName() {
 *   const record = useViewerRecord('basicProfile')
 *
 *   return (
 *     <button
 *       disabled={!record.isMutable || record.isMutating}
 *       onClick={async () => {
 *         await record.merge({ name: 'Alice' })
 *       }}>
 *       Set name
 *     </button>
 *   )
 * }
 * ```
 *
 * ### Read a public record
 *
 * The {@linkcode react.usePublicRecord usePublicRecord} hook is similar to the
 * `useViewerRecord` hook described above, but reading from the index of an explicitly provided
 * account rather than the viewer. Public records are read-only, `useViewerRecord` must be used in
 *  case mutations are needed.
 *
 * ```ts
 * import { usePublicRecord } from '@self.id/framework'
 *
 * function ShowProfileName({ did }) {
 *   const record = usePublicRecord('basicProfile', did)
 *
 *   const text = record.isLoading
 *     ? 'Loading...'
 *     : record.content
 *     ? `Hello ${record.content.name || 'stranger'}`
 *     : 'No profile to load'
 *   return <p>{text}</p>
 * }
 * ```
 *
 * ### Server-side prefetching
 *
 * Server-side rendering can be used to improve the user experience for the first load of an app or
 * page. The framework exports a {@linkcode react.RequestClient RequestClient} class from the
 * {@linkcode react @self.id/react} package that can be used to fetch wanted records on the server
 * in order to have them immediately available by the `usePublicRecord` and `useViewerRecord`
 * hooks.
 *
 * The following example shows how this can be used in a [Next.js](https://nextjs.org/) page, using
 * the `ShowViewerName` component created in the previous example:
 *
 * ```ts
 * import { Provider, RequestClient } from '@self.id/framework'
 *
 * export const getServerSideProps = async (ctx) => {
 *   const client = new RequestClient({
 *     ceramic: 'testnet-clay',
 *     // Inject the cookie from the request headers to parse the viewerID
 *     cookie: ctx.req.headers.cookie,
 *   })
 *   if (client.viewerID != null) {
 *     // If the viewerID is set, fetch its profile
 *     await client.prefetch('basicProfile', client.viewerID)
 *   }
 *   return { props: { state: client.getState() } }
 * }
 *
 * // Use the state prop injected by the server
 * export default function Home({ state }) {
 *   return (
 *     <Provider state={state}>
 *       <ShowViewerName />
 *     </Provider>
 *   )
 * }
 * ```
 *
 * ## Re-exported classes
 *
 * - {@linkcode core.Core}
 * - {@linkcode core.PublicID}
 * - {@linkcode react.ReactClient}
 * - {@linkcode react.RequestClient}
 * - {@linkcode web.SelfID}
 * - `EthereumAuthProvider` from 3ID Connect
 *
 *
 * @module framework
 */

// DataModels re-exports
export type { Account as AlsoKnownAsAccount, AlsoKnownAs } from '@datamodels/identity-accounts-web'
export type { BasicProfile } from '@datamodels/identity-profile-basic'

// SDK re-exports ignored from docs to avoid duplications (links added above)
/** @ignore */
export { Core, PublicID } from '@self.id/core'
/** @ignore */
export { ReactClient, RequestClient } from '@self.id/react'
/** @ignore */
export { EthereumAuthProvider, SelfID } from '@self.id/web'

// Documented SDK re-exports
export { isCAIP10string, isDIDstring } from '@self.id/core'
export { selectImageSource, uploadImage } from '@self.id/image-utils'
export type { Dimensions, ImageSources } from '@self.id/image-utils'
export {
  Provider,
  ViewerConnectedContainer,
  useClient,
  usePublicRecord,
  useViewerConnection,
  useViewerID,
  useViewerRecord,
} from '@self.id/react'
export type {
  ProviderProps,
  PublicRecord,
  RequestClientParams,
  RequestState,
  ViewerConnectedContainerProps,
  ViewerConnectionState,
  ViewerRecord,
} from '@self.id/react'
export type { ConnectNetwork, EthereumProvider } from '@self.id/web'

// Local exports
export { formatDID, getImageURL } from './utils.js'
