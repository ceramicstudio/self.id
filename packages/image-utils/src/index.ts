/**
 * Image utilities for Self.ID profiles.
 *
 * ## Purpose
 *
 * The `image-utils` module provides various utility functions for images, such as selecting the
 * most suitable format among sources, resizing and uploading, and is mostly meant to be used in a
 * Web browser context.
 *
 * ## Installation
 *
 * ```sh
 * npm install @self.id/image-utils
 * ```
 *
 * ## Common use-cases
 *
 * ### Resize and upload an image to multiple dimensions
 *
 * Using a {@linkcode web.SelfID SelfID} instance from the {@linkcode web} module.
 *
 * ```ts
 * import { uploadImage } from '@self.id/image-utils'
 *
 * // The `file` argument must implement the File interface from
 * // https://developer.mozilla.org/en-US/docs/Web/API/File
 * async function setProfileImage(selfID, file) {
 *   const imageSources = await uploadFile(
 *     'https://ipfs.infura.io:5001/api/v0',
 *     file,
 *     [{ width: 60, height: 60 }, { width: 200, height: 200 }],
 *   )
 *   // Here `selfID` must be an instance of `SelfID` from the `web` module
 *   await selfID.merge({ image: imageSources })
 * }
 * ```
 *
 * ### Get the most suitable image source for given dimensions
 *
 * Using a {@linkcode core.Core Core} instance from the {@linkcode core} module.
 *
 * ```ts
 * import { Core } from '@self.id/core'
 * import { selectImageSource } from '@self.id/image-utils'
 *
 * const core = new Core({ ceramic: 'testnet-clay' })
 *
 * async function getAvatarSource(did) {
 *   const profile = await core.get('basicProfile', did)
 *   return profile?.image ? selectImageSource(profile.image, { width: 60, height: 60 }) : null
 * }
 * ```
 *
 * @module image-utils
 */

export * from './ipfs.js'
export * from './loading.js'
export * from './selection.js'
export * from './types.js'
