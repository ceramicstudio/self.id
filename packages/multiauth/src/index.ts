/**
 * Blockchain authentication for React apps.
 *
 * ## Purpose
 *
 * The `multiauth` module provides a React Provider component and hook to access a Wallet provider
 * from a given Wallet, notably useful in order to authenticate a DID for creating a
 * {@linkcode web.SelfID SelfID} instance.
 *
 * ## Installation
 *
 * The `multiauth` module uses components from the [Grommet library](https://v2.grommet.io/), that
 * needs to be explicitly installed along `@self.id/multiauth`:
 *
 * ```sh
 * npm install @self.id/multiauth grommet
 * ```
 *
 * ## Common use-cases
 *
 * ### Add the Provider to your component tree
 *
 * As the `multiauth` module uses components from the [Grommet library](https://v2.grommet.io/), a
 * top-level container must be added as a parent of the {@linkcode multiauth.Provider} component.
 *
 * The {@linkcode ui.Provider Provider} component from the {@linkcode ui} module can be used to
 * inject this container with a theme matching the [Self.ID application](https://self.id). In such
 * cases, the {@linkcode ui @self.id/ui} module needs to also be installed.
 *
 * ```ts
 * import { Provider as AuthProvider } from '@self.id/multiauth'
 * import { Provider as UIProvider } from '@self.id/ui'
 *
 * const networks = [
 *   {
 *     key: 'ethereum',
 *     connectors: [
 *       { key: 'injected' },
 *       { key: 'fortmatic', params: { apiKey: fortmaticApiKey }},
 *     ],
 *   },
 * ]
 *
 * function App({ children }) {
 *   return (
 *     <UIProvider>
 *       <AuthProvider networks={networks}>{children}</AuthProvider>
 *     </UIProvider>
 *   )
 * }
 * ```
 *
 * ### Access the authentication state
 *
 * ```ts
 * import { useMultiAuth } from '@self.id/multiauth'
 *
 * function useWallet() {
 *   const [authState, authenticate] = useMultiAuth()
 *   return [authState.status === 'authenticated' ? authState.auth : null, authenticate]
 * }
 * ```
 *
 * @module multiauth
 */

export type { ModalConfig } from './components/Modal'
export { Provider } from './components/Provider'
export type {
  ProviderConfig as MultiAuthProviderConfig,
  ProviderProps,
} from './components/Provider'
export * from './connectors'
export * from './hooks'
export * from './networks'
export * from './providers/types'
export * from './types'
