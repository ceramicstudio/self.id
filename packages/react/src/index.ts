/**
 * React hooks and utilities for authentication and records interactions.
 *
 * ## Purpose
 *
 * The `react` module provides React components, hooks and related utility functions to help manage
 * authentication and interactions with records, similar to the {@linkcode web} module but designed
 * specifically to be used with React.
 *
 * ## Installation
 *
 * ```sh
 * npm install @self.id/react
 * ```
 *
 * ## Common use-cases
 *
 * ### Configure the Provider
 *
 * The {@linkcode Provider} component must be added at the root of the
 * application tree in order to use the hooks described below. It can be used to provide a custom
 * configuration for the Self.ID clients and queries, as well as initial state.
 *
 * ```ts
 * import { Provider } from '@self.id/react'
 *
 * function App({ children }) {
 *   return <Provider client={{ ceramic: 'testnet-clay' }}>{children}</Provider>
 * }
 * ```
 *
 * ### Authenticate the user
 *
 * The module provides a React hook to easily initiate an authentication flow for the Viewer
 * (the "current user" of the app) using an `EthereumAuthProvider` instance, notably exported by
 * the {@linkcode web} module. Once authenticated, the Viewer's cookie is set to the authenticated
 * DID and writing records associated to the Viewer becomes possible.
 *
 * ```ts
 * import { useViewerConnection } from '@self.id/react'
 * import { EthereumAuthProvider } from '@self.id/web'
 *
 * async function createAuthProvider() {
 *   // The following assumes there is an injected `window.ethereum` provider
 *   const addresses = await window.ethereum.request({ method: 'eth_requestAccounts' })
 *   return new EthereumAuthProvider(window.ethereum, addresses[0])
 * }
 *
 * // A simple button to initiate the connection flow. A Provider must be present at a higher level
 * // in the component tree for the `useViewerConnection()` hook to work.
 * function ConnectButton() {
 *   const [connection, connect, disconnect] = useViewerConnection()
 *
 *   return connection.status === 'connected' ? (
 *     <button
 *       onClick={() => {
 *         disconnect()
 *       }}>
 *       Disconnect ({connection.selfID.id})
 *     </button>
 *   ) : 'ethereum' in window ? (
 *     <button
 *       disabled={connection.status === 'connecting'}
 *       onClick={() => {
 *         createAuthProvider().then(connect)
 *       }}>
 *       Connect
 *     </button>
 *   ) : (
 *     <p>
 *       An injected Ethereum provider such as{' '}
 *       <a href="https://metamask.io/">MetaMask</a> is needed to authenticate.
 *     </p>
 *   )
 * }
 * ```
 *
 * ### Read a viewer record
 *
 * The {@linkcode useViewerRecord} hook loads the record for a given
 * definition in the index of the current viewer, with the following variants:
 *
 * - If no viewer is set, no record can be loaded
 * - If the viewer is not authenticated, the record gets loaded but cannot be mutated
 * - If the viewer is authenticated, the record gets loaded and be mutated
 *
 * ```ts
 * import { useViewerRecord } from '@self.id/react'
 *
 * function ShowViewerName() {
 *   const record = useViewerRecord('basicProfile')
 *
 *   const text = record.isLoading
 *     ? 'Loading...'
 *     : record.content
 *     ? `Hello ${record.content.name || 'stranger'}`
 *     : 'No profile to load'
 *   return <p>{text}</p>
 * }
 * ```
 *
 * ### Read a public record
 *
 * The {@linkcode usePublicRecord} hook is similar to the
 * `useViewerRecord` hook described above, but reading from the index of an explicitly provided
 * account rather than the viewer. Public records are read-only, `useViewerRecord` must be used in
 *  case mutations are needed.
 *
 * ```ts
 * import { usePublicRecord } from '@self.id/react'
 *
 * function ShowProfileName({ did }) {
 *   const record = usePublicRecord('basicProfile', did)
 *
 *   const text = record.isLoading
 *     ? 'Loading...'
 *     : record.content
 *     ? `Hello ${record.content.name || 'stranger'}`
 *     : 'No profile to load'
 *   return <p>{text}</p>
 * }
 * ```
 *
 * ### Server-side prefetching
 *
 * Server-side rendering can be used to improve the user experience for the first load of an app or
 * page. The module exports a {@linkcode RequestClient} class that can be used
 * to fetch wanted records on the server in order to have them immediately available by the
 * `usePublicRecord` and `useViewerRecord` hooks.
 *
 * The following example shows how this can be used in a [Next.js](https://nextjs.org/) page, using
 * the `ShowViewerName` component created in the previous example:
 *
 * ```ts
 * import { Provider, RequestClient } from '@self.id/react'
 *
 * export const getServerSideProps = async (ctx) => {
 *   const client = new RequestClient({
 *     ceramic: 'testnet-clay',
 *     // Inject the cookie from the request headers to parse the viewerID
 *     cookie: ctx.req.headers.cookie,
 *   })
 *   if (client.viewerID != null) {
 *     // If the viewerID is set, fetch its profile
 *     await client.prefetch('basicProfile', client.viewerID)
 *   }
 *   return { props: { state: client.getState() } }
 * }
 *
 * // Use the state prop injected by the server
 * export default function Home({ state }) {
 *   return (
 *     <Provider state={state}>
 *       <ShowViewerName />
 *     </Provider>
 *   )
 * }
 * ```
 *
 * @module react
 */

export * from './components/Provider.js'
export * from './components/ViewerConnectedContainer.js'
export * from './constants.js'
export * from './client.js'
export * from './hooks.js'
export * from './server.js'
export * from './state.js'
export * from './types.js'
