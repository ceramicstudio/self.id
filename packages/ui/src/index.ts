/**
 * UI
 *
 * ```sh
 * npm install @self.id/ui
 * ```
 *
 * @module ui
 */

export { AvatarPlaceholder } from './components/AvatarPlaceholder'
export type { AvatarPlaceholderProps } from './components/AvatarPlaceholder'
export { Provider } from './components/Provider'
export type { ProviderProps } from './components/Provider'
export { colors, theme } from './theme'
export type { Colors, ColorType, ThemeType } from './theme'
