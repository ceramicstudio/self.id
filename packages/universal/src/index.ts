export * from './config'
export * from './constants'
export * from './core'
export * from './images'
export * from './public'
export * from './types'
