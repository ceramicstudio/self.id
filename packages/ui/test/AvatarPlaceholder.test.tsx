/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react'

import { AvatarPlaceholder } from '../src'

describe('AvatarPlaceholder', () => {
  test('uses the provided did and size props', () => {
    const { asFragment } = render(<AvatarPlaceholder did="did:test:123" size={32} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('uses a default value when the did prop is not provided', () => {
    const { asFragment } = render(<AvatarPlaceholder size={20} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
