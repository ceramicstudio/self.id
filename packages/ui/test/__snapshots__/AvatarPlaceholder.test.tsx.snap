// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarPlaceholder uses a default value when the did prop is not provided 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="20"
    role="img"
    viewBox="0 0 80 80"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      self.id
    </title>
    <mask
      height="80"
      id="mask__marble"
      maskUnits="userSpaceOnUse"
      width="80"
      x="0"
      y="0"
    >
      <rect
        fill="#FFFFFF"
        height="80"
        rx="160"
        width="80"
      />
    </mask>
    <g
      mask="url(#mask__marble)"
    >
      <rect
        fill="#228DFF"
        height="80"
        width="80"
      />
      <path
        d="M32.414 59.35L50.376 70.5H72.5v-71H33.728L26.5 13.381l19.057 27.08L32.414 59.35z"
        fill="#BA01FF"
        filter="url(#prefix__filter0_f)"
        transform="translate(-2 2) rotate(-226 40 40) scale(1.5)"
      />
      <path
        d="M22.216 24L0 46.75l14.108 38.129L78 86l-3.081-59.276-22.378 4.005 12.972 20.186-23.35 27.395L22.215 24z"
        fill="#FF0092"
        filter="url(#prefix__filter0_f)"
        style="mix-blend-mode: overlay;"
        transform="translate(7 7) rotate(159 40 40) scale(1.5)"
      />
    </g>
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="userSpaceOnUse"
        id="prefix__filter0_f"
      >
        <feflood
          flood-opacity="0"
          result="BackgroundImageFix"
        />
        <feblend
          in="SourceGraphic"
          in2="BackgroundImageFix"
          result="shape"
        />
        <fegaussianblur
          result="effect1_foregroundBlur"
          stdDeviation="7"
        />
      </filter>
    </defs>
  </svg>
</DocumentFragment>
`;

exports[`AvatarPlaceholder uses the provided did and size props 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="32"
    role="img"
    viewBox="0 0 80 80"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      did:test:123
    </title>
    <mask
      height="80"
      id="mask__marble"
      maskUnits="userSpaceOnUse"
      width="80"
      x="0"
      y="0"
    >
      <rect
        fill="#FFFFFF"
        height="80"
        rx="160"
        width="80"
      />
    </mask>
    <g
      mask="url(#mask__marble)"
    >
      <rect
        fill="#228DFF"
        height="80"
        width="80"
      />
      <path
        d="M32.414 59.35L50.376 70.5H72.5v-71H33.728L26.5 13.381l19.057 27.08L32.414 59.35z"
        fill="#BA01FF"
        filter="url(#prefix__filter0_f)"
        transform="translate(-6 -6) rotate(-46 40 40) scale(1.3)"
      />
      <path
        d="M22.216 24L0 46.75l14.108 38.129L78 86l-3.081-59.276-22.378 4.005 12.972 20.186-23.35 27.395L22.215 24z"
        fill="#FF0092"
        filter="url(#prefix__filter0_f)"
        style="mix-blend-mode: overlay;"
        transform="translate(-1 1) rotate(-249 40 40) scale(1.3)"
      />
    </g>
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="userSpaceOnUse"
        id="prefix__filter0_f"
      >
        <feflood
          flood-opacity="0"
          result="BackgroundImageFix"
        />
        <feblend
          in="SourceGraphic"
          in2="BackgroundImageFix"
          result="shape"
        />
        <fegaussianblur
          result="effect1_foregroundBlur"
          stdDeviation="7"
        />
      </filter>
    </defs>
  </svg>
</DocumentFragment>
`;
